signer/src/block_observer.rs

```diff
@@ -187,24 +184,41 @@ where
         Ok(())
     }
 
+    /// Fetch deposit requests from Emily and store the validated ones into
+    /// the database.
     #[tracing::instrument(skip(self))]
     async fn load_latest_deposit_requests(&mut self) -> Result<(), Error> {
-        let deposit_requests = self.emily_client.get_deposits().await?;
+        let mut deposit_requests = Vec::new();
+        let mut failed_requests = Vec::new();
 
-        for request in deposit_requests {
+        for request in self.emily_client.get_deposits().await? {
             let deposit = request
                 .validate(&self.context.get_bitcoin_client())
                 .await
                 .inspect_err(|error| tracing::warn!(%error, "could not validate deposit request"));
 
-            if let Ok(Some(deposit)) = deposit {
-                self.deposit_requests
-                    .entry(deposit.info.outpoint.txid)
-                    .or_default()
-                    .push(deposit);
+            match deposit {
+                // Happy path, we've successfully validated the deposit
+                // request.
+                Ok(Some(deposit)) => deposit_requests.push(deposit),
+                // This happens when the transaction has failed the first
+                // step of validation or has passed the first step but we
+                // don't recongnize the x-only public key in the deposit
+                // script.
+                Err(Error::SbtcLib(_)) | Err(Error::UnknownAggregateKey(_, _)) => {
+                    failed_requests.push(request);
```

Comment:
I'm not entirely sure what, if anything, we should do with these failed requests. Should we tell Emily, persist them to the database, store them in memory but with a capped size, or do nothing?